src/tests/test_affiliate/WidgetsTest.php

```diff
@@ -19,23 +19,134 @@ class WidgetsTest extends TestCase {
      * Note that these tests will fail for non-Gutenberg WordPress (< version 5).
      */
 
+    // Keep track of posts created during the tests so that we can delete them on tear down.
+    static $postIDsToDelete = [];
+
+    static function tearDownAfterClass(): void
+    {
+        $browser = SeleniumBrowser::getTestBrowser();
+        try {
+            $browser->deletePosts( WidgetsTest::$postIDsToDelete );
+        } catch ( Exception $e ) {
+            error_log( 'Failed to delete test posts: ' . $e->getMessage() );
+        } finally {
+            $browser->quit();
+        }
+    }
+
     function wordPressVersionTooLow() : bool {
         if ( !empty( WP_VERSION ) && intval( substr( WP_VERSION, 0, 1 ) ) < 5 ) {
             return true;
         }
         return false;
     }
 
-    function checkWidgetIsAvailable( $blockType ) {
+    /**
+     * When the user selects an Organic Affiliate widgets block, an IFrame with login fields will appear.
+     * This function logs us in as the test user so that we can customize and insert the widget.
+     * @param SeleniumBrowser $browser
+     * @return void
+     * @throws Exception
+     */
+    private function logIntoWidgetsSelectionIFrame( SeleniumBrowser $browser ) {
+        $iframe = $browser->getOrganicIframe();
+        $browser->switchToIframe( $iframe );
+        // Log in with test account.
+        $browser->fillTextInput( '#email', ORGANIC_TEST_USER_EMAIL );
+        $browser->fillTextInput( '#password', ORGANIC_TEST_USER_PASSWORD );
+        $browser->click( '#signin-button' );
+    }
+
+    /**
+     * Search for and select the Test Product in our Organic Affiliate widgets IFrame.
+     * @param SeleniumBrowser $browser
+     * @return void
+     * @throws Exception
+     */
+    private function selectTestProduct( SeleniumBrowser $browser ) {
+        $test_product_guid = TEST_PRODUCT_GUID;
+        $browser->fillTextInput( '#affiliate-product-search-entry', 'Selenium Test Product' );
+        try {
+            $browser->click("[data-test-element=\"affiliate-product-select-{$test_product_guid}\"]" );
```

Comment:
I will add a space after `click(`, but want to do so after soliciting feedback to avoid kicking off GitHub Actions for such a minor change.